pytorch_lightning/profiler/pytorch.py

```diff
@@ -237,7 +200,8 @@ def on_train_start(self, local_rank: Optional[str] = None, log_dir: str = None):
         streaming_out = [self.output_file.write] if self.output_file else [log.info]
         super().__init__(output_streams=streaming_out)
 
-    def wrap_functions_into_rank_zero_only(self):
+    def wrap_functions_into_rank_zero_only(self) -> None:
+        # TODO: what's the point of wrapping like this?
```

Comment:
Address this

Reply:
The function should be run only on rank zero if `output_fname` is not profiled. 
If running with DDP and the user provides `output_fname`, each process will save their own profiling into `output_fname_{rank}.txt`

Reply:
```suggestion
```